src/useQuery.js

```diff
@@ -15,9 +15,13 @@ module.exports = function useQuery(query, opts = {}) {
     ...opts
   });
 
+  // TODO revise this logic
```

Comment:
Yea I think we can first check if `state.data || state.error` exists instead - if the query was already SSR'd then the initialState will be the cacheHit.